status-im/status-react
# Upgrade react native to 0.53.3

[comment]: # (Please replace ... with your information. Remove < and >)
[comment]: # (To auto-close issue on merge, please insert the related issue number after # i.e fixes #566)

Fixes #3377

### Summary:

[comment]: # (Summarise the problem and how the pull request solves it)

Upgrades:
- React Native to 0.53.3
- Re-Natal to 0.8.1 (required for changes in metro-bundler -> metro)

### Review notes:

Includes workarounds for [react-native-vector-icons (or rather, Metro)](https://github.com/oblador/react-native-vector-icons/issues/626#issuecomment-362386341) and [FlatList keyExtractor](https://github.com/wonday/react-native-pdf/issues/125).

As a side note, we could probably safely upgrade to re-frame 0.10.2 -> 0.10.4 which is what [Re-Natal 0.8.1 is on](https://github.com/drapanjanas/re-natal/releases/tag/v0.8.1).

### Steps to test:

- `rm -rf node_modules`
- Rebuild according to [Wiki](https://wiki.status.im/Building_Status)
- Confirm nothing is obviously broken in app
- Confirm tests pass (with the exception of `group-transactions-by-date`, which also fails in develop, though I suspect the tests do not actually test React to begin with)

status: ready
